ui/src/dataExplorer/components/SaveAsTaskForm.tsx

```diff
@@ -109,14 +103,58 @@ class SaveAsTaskForm extends PureComponent<Props & WithRouterProps> {
           onSubmit={this.handleSubmit}
           canSubmit={this.isFormValid}
           dismiss={dismiss}
-          tokens={tokens}
+          tokens={this.getRelevantTokens()}
           selectedToken={selectedToken}
           onTokenChange={this.handleTokenChange}
         />
       </SpinnerContainer>
     )
   }
 
+  private getRelevantTokens() {
+    const authorizations = this.props.tokens
+
+    const {draftQueries, activeQueryIndex} = this.props
+    let readBucketName = ''
+    if (draftQueries[activeQueryIndex].editMode === 'builder') {
```

Comment:
Instead of re-indexing on `draftQueries[activeQueryIndex]`, why not story that query in a variable i.e.

`const query = draftQuery[activeQueryIndex]`
 
then you can:

`query.editMode`
`query.text`

etc